backend/contracts/Quiz.sol

```diff
@@ -3,16 +3,19 @@ pragma solidity ^0.8.0;
 
 import "@oasisprotocol/sapphire-contracts/contracts/Sapphire.sol";
 import {EIP155Signer} from "@oasisprotocol/sapphire-contracts/contracts/EIP155Signer.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+// import "@openzeppelin/contracts/access/Ownable.sol";
 
-contract Quiz {
+
+contract Quiz is ERC721 {
```

Comment:
Let's decouple Quiz and ERC721. Add `setNft()` (and  `getNft()`) with parameters like the NFT name, symbol and image URI which will instantiate and reference a new ERC721. Then in the claimReward(), the function checks, whether the NFT was initialized and performs the NFT minting.

Reply:
Sure, added separate contract for the ERC721, now stores URIs at constructor level.